pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -871,3 +887,25 @@ func cephVolumeEnvVar() []v1.EnvVar {
 		{Name: "CEPH_VOLUME_SKIP_RESTORECON", Value: "1"},
 	}
 }
+
+func (c *Cluster) osdRunFlagTuningOnPVC(osdID int, storageClassName string) error {
+	who := fmt.Sprintf("osd.%d", osdID)
+	do := make(map[string]string)
+
+	if storageClassName == "gp2" {
```

Comment:
How can we make this more general than just for gp2? We can't predict what other storage classes will also have the same issue as gp2 so I was thinking we would need to expose this as a setting in the storageClassDeviceSet as suggested in #4298.

Reply:
The point is not to make it general :) but on per case basis, at least for now based on what we saw downstream.

What would you like to expose in the CRD? Adding a `deviceClass` to the CRD is not helping here and might just create more confusion as suggested by @jdurgin.

Reply:
Perhaps in the long term ceph can tune itself better based on testing drive performance - this can be difficult with sometimes drastic differences after hours of use, so some continuous tuning may be needed - this ties in well with rados QoS long term as well.

For now though, I don't see a better way to do this - ceph has no idea what gp2 is, and can't tell a particular drive is gp2 from block device. At some level it seems we need to add awareness of this, and in the short term rook seems like a reasonable place to put it.

Reply:
What I'm looking for is a way to categorize this into something that can be exposed in the CR. For example, should we have a setting like `tuneDeviceBecauseItIsntReallyAsFastAsAnSSD`? That name is of course ridiculous, but if we can come up with a name for this behavior, we can expose it in the CR. I don't think we should only do this for gp2.

Reply:
Thoughts: `assumeSlowDevice: true`, `devicePerformanceClass: "spinning"`, .... that's all I've got right now.

Reply:
I'm hesitating to add a new CR property for this. In the end, you can simply tune your OSDs using a `rook-config-override` confimap if something is not right...
Currently, we are only seeing this with GP2, probably because that's the only thing that has been tested so far.

The auto-detection can be done by Rook for now and maybe later using the same mechanism in Ceph by checking an env?

Reply:
Right, we have only tested on gp2. If we had tested more broadly and still only saw gp2 with the issue, then maybe we could approach it like this. Until then it seems we have a more general problem that we could solve by leaving it up to the admin to decide with a CR setting.

Reply:
I still believe anyone can solve this by either:

* using `rook-config-override`
* use `ceph config set`

Obviously this requires some knowledge on which flag to set, with which value.
Based on that I see the value of a CR setting.